Quaver.Shared/Graphics/Containers/PoolableScrollContainer.cs

```diff
@@ -230,17 +230,42 @@ protected PoolableSprite<T> AddObject(int index, bool updateContent = true)
             {
                 var drawable = CreateObject(AvailableItems[index], index);
                 drawable.DestroyIfParentIsNull = false;
-                drawable.Y = (PoolStartingIndex + index) * drawable.Height + PaddingTop;
 
-                if (updateContent)
-                    drawable.UpdateContent(AvailableItems[index], index);
+                Pool.Insert(index, drawable);
 
-                Pool.Add(drawable);
+                if (updateContent)
+                {
+                    for (int i = index; i < Pool.Count; i++)
+                    {
+                        var foo = Pool[i];
+                        foo.Y = (PoolStartingIndex + i) * foo.Height + PaddingTop;
+                        foo.UpdateContent(AvailableItems[i], i);
+                    }
+                }
 
                 return drawable;
             }
         }
 
+        protected void AddObjectAtIndex(int index, T obj, bool scrollTo, bool usePoolCount = false)
+        {
+            lock (AvailableItems)
+            lock (Pool)
+            {
+                if (!AvailableItems.Contains(obj))
+                    AvailableItems.Insert(index, obj);
+
+                // Need another drawable to use
+                if (Pool.Count < PoolSize)
+                    AddContainedDrawable(AddObject(index));
+
+                RecalculateContainerHeight(usePoolCount);
+
+                if (scrollTo)
+                    ScrollTo(-index * DrawableHeight, 1000);
+            }
+        }
```

Comment:
This portion causes song select as well as the jukebox to crash the game.

```
[12:39:53] - RUNTIME - ERROR: System.ArgumentOutOfRangeException: Index must be within the bounds of the List. (Parameter 'index')
   at System.Collections.Generic.List`1.Insert(Int32 index, T item)
   at Quaver.Shared.Graphics.Containers.PoolableScrollContainer`1.AddObject(Int32 index, Boolean updateContent) in C:\Users\Swan\Desktop\git\Quaver\Quaver.Shared\Graphics\Containers\PoolableScrollContainer.cs:line 234
   at Quaver.Shared.Graphics.Containers.PoolableScrollContainer`1.CreatePool(Boolean containDrawables, Boolean updateContent) in C:\Users\Swan\Desktop\git\Quaver\Quaver.Shared\Graphics\Containers\PoolableScrollContainer.cs:line 119
   at Quaver.Shared.Screens.Selection.UI.Mapsets.SongSelectContainer`1..ctor(List`1 availableItems, Int32 poolSize) in C:\Users\Swan\Desktop\git\Quaver\Quaver.Shared\Screens\Selection\UI\Mapsets\SongSelectContainer.cs:line 86
   at Quaver.Shared.Screens.Selection.UI.Mapsets.MapsetScrollContainer..ctor(Bindable`1 availableMapsets, Bindable`1 activeScrollContainer) in C:\Users\Swan\Desktop\git\Quaver\Quaver.Shared\Screens\Selection\UI\Mapsets\MapsetScrollContainer.cs:line 72
   at Quaver.Shared.Screens.Selection.SelectionScreenView.CreateMapsetContainer() in C:\Users\Swan\Desktop\git\Quaver\Quaver.Shared\Screens\Selection\SelectionScreenView.cs:line 297
   at Quaver.Shared.Screens.Selection.SelectionScreenView..ctor(SelectionScreen screen) in C:\Users\Swan\Desktop\git\Quaver\Quaver.Shared\Screens\Selection\SelectionScreenView.cs:line 119
   at Quaver.Shared.Screens.Selection.SelectionScreen..ctor() in C:\Users\Swan\Desktop\git\Quaver\Quaver.Shared\Screens\Selection\SelectionScreen.cs:line 136
   at Quaver.Shared.Screens.Main.MainMenuScreen.<>c.<ExitToSinglePlayer>b__21_2() in C:\Users\Swan\Desktop\git\Quaver\Quaver.Shared\Screens\Main\MainMenuScreen.cs:line 175
   at Quaver.Shared.Screens.QuaverScreenManager.<>c__DisplayClass24_0.<ScheduleScreenChange>b__0() in C:\Users\Swan\Desktop\git\Quaver\Quaver.Shared\Screens\QuaverScreenManager.cs:line 205
   at Quaver.Shared.Scheduling.ThreadScheduler.<>c__DisplayClass1_0.<Run>b__0() in C:\Users\Swan\Desktop\git\Quaver\Quaver.Shared\Scheduling\ThreadScheduler.cs:line 38```

Reply:
I'm not exactly confident in the fix, but nothing seems to be broken